src/main/java/LinkedBinarySearchTree.java

```diff
@@ -63,43 +63,39 @@ public boolean remove(T element) {
     /**
      * Helper method for recursive remove.
      *
-     * @param element Element being searched for for removal
-     * @param parent  The parent of the node being investigated
-     * @param child   The node being investigated
-     * @return The element being removed
+     * @param element Element to me removed.
+     * @param parent  The parent of the current node.
+     * @param current The current node.
+     * @return The element being removed.
      */
-    private T remove(T element, Node<T> parent, Node<T> child) {
-        if (child == null) {
-            throw new NoSuchElementException();
-        } else if (child.getData().equals(element)) {
-            if (parent.getData().compareTo(element) > 0) {
-                parent.setLeft(findReplacementNode(child));
+    private T remove(T element, Node<T> parent, Node<T> current) {
+        if (current == null) {
+            throw new NoSuchElementException("Empty Tree");
+        } else if (Objects.equals(current.getData(), element)) {
+            if (Objects.compare(parent.getData(), current.getData(), T::compareTo) > 0) {
```

Comment:
Changed the comparison to parent data vs. current data from parent data vs. element since, although element and the child data would be equal here, it's the child data that is more _meaningful_ in the comparison.